hstream/client/Main.hs

```diff
@@ -0,0 +1,146 @@
+{-# LANGUAGE DeriveAnyClass      #-}
+{-# LANGUAGE DeriveGeneric       #-}
+{-# LANGUAGE LambdaCase          #-}
+{-# LANGUAGE OverloadedStrings   #-}
+{-# LANGUAGE RecordWildCards     #-}
+{-# LANGUAGE ScopedTypeVariables #-}
+{-# LANGUAGE TypeApplications    #-}
+
+module Main where
+
+import           Conduit
+import           Control.Exception        (SomeException, try)
+import           Data.Aeson               (FromJSON, eitherDecode')
+import qualified Data.ByteString.Char8    as BC
+import qualified Data.ByteString.Lazy     as BL
+import qualified Data.List                as L
+import           Data.Proxy               (Proxy (..))
+import           Data.Text                (pack)
+import           HStream.SQL
+import           HStream.Server.Type
+import           Network.HTTP.Simple      (Request, getResponseBody, httpBS,
+                                           httpSink, parseRequest,
+                                           setRequestBodyJSON, setRequestMethod)
+import           Options.Applicative      (Parser, auto, execParser, fullDesc,
+                                           help, helper, info, long, metavar,
+                                           option, progDesc, short, showDefault,
+                                           strOption, value, (<**>))
+import           System.Console.Haskeline (Completion, CompletionFunc, InputT,
+                                           Settings, completeWord,
+                                           defaultSettings, getInputLine,
+                                           handleInterrupt, runInputT,
+                                           setComplete, simpleCompletion,
+                                           withInterrupt)
+import           Text.Pretty.Simple       (pPrint)
+
+parseConfig :: Parser ClientConfig
+parseConfig =
+  ClientConfig
+    <$> strOption (long "url" <> metavar "string" <> showDefault <> value "http://localhost" <> short 'b' <> help "base url valur")
+    <*> option auto (long "port" <> showDefault <> value 8081 <> short 'p' <> help "port value" <> metavar "INT")
+
+def :: Settings IO
+def = setComplete compE defaultSettings
+
+compE :: CompletionFunc IO
+compE = completeWord Nothing [] compword
+
+-- should make sure there is no empty command
+wordTable :: [[String]]
+wordTable =
+  [ ["show", "querys"],
+    ["delete", "query"],
+    ["delete", "query", "all"],
+    [":h"],
+    [":q"]
+  ]
+
+-- for complete wordTable command
+generalComplete :: [[String]] -> [String] -> [String]
+generalComplete t [] = L.nub (map head t)
+generalComplete t [x] = case L.nub (filter (L.isPrefixOf x) (map head t)) of
+  [w]
+    | x == w ->
+      map (\z -> x ++ " " ++ z) (generalComplete (filter (/= []) (map tail (filter (\z -> head z == x) t))) [])
+  ws -> ws
+generalComplete t (x : xs) =
+  map (\z -> x ++ " " ++ z) (generalComplete (filter (/= []) (map tail (filter (\z -> head z == x) t))) xs)
+
+-- for complete dbid & tbid
```

Comment:
Make sure your sentence is correct grammatically. And what does `dbid` and `tbid` mean?